dvc/command/add.py

```diff
@@ -74,6 +78,35 @@ def add_parser(subparsers, parent_parser):
         help="Specify name of the DVC-file this command will generate.",
         metavar="<filename>",
     )
+    parser.add_argument(
+        "-o",
+        "--out",
+        help="Destination path to put files to.",
+        metavar="<path>",
```

Comment:
Now that I think about it. I don't see why --out relates especially to transferring data straight to remote. Can you remind me why this is needed or desired please @isidentical ?

Reply:
@jorgeorpinel you need to fill in the `out` path in the .dvc files that is being created. When you do the regular `dvc add something`. `something` becomes that value. In this case since data is remote you need to specify the way it should be "mapped" to the project's tree.

Reply:
@shcheklein but you still need to specify a target e.g. `add /tmp/fileordir --to-remote`, so it's possible to use the base name (`fileordir` in this case). That's how the default value for `out` is picked for `import-url`, for example. That's why I find it strange that to-remote requires an explicit output path.

I think `add --out` is a completely separate feature. BTW I would agree on introducing it (but not just for to-remote cases — that's the strange part IMO).

Reply:
`import-url` has `[out]` for the same reason - to specify where do you want to put that data. It's like with `wget`, `curl`, etc - URLs come often in some raw format that you don't want to use in the workspace. For usual `dvc add` you don't need it, data comes from the workspace already.

If we by default use the same logic as in `import-url` to determine names, then this feature makes sense to my mind and more or less cosistent.

Reply:
> If we by default use the same logic as in import-url to determine names, then this feature makes sense

🤝 

WDYT @isidentical ?

Reply:
Just to clarify, we still keep `--out`, we make it (keep it) as not required.

Reply:
If I am not misunderstanding, it already works in that way. If you don't pass something with `--out`, it will fall back to the strategy that import-url uses.